### PR TITLE
Inra inaro suggestion 90

### PR DESCRIPTION
**Pull request by InraInaro**

### CODE REVIEW

**Comment by aika092** [comment]
There are some bugs in this but I will fix them on my end.

**Comment by InraInaro** [comment]
If able, may you inform me on what's causing the bugs? I'm curious about what I missed or got wrong.

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 1]
```diff
@@ -207,6 +207,54 @@ To compute stat healing:
 		decrease temp_int_dam by 1;
 		say "[bold type]You feel as though your mind is slightly clearer![roman type]";
 
+A time based rule (this is the audible jiggling rule):
+	if a random number between 1 and 100 < 50 + unlucky:
+		if debugmode > 0, say "AUDIBLE CHECK. ";
+		let audibly jiggling clothing be nothing;
+		repeat with C running through worn cursed blandness clothing:
+			if C is overdress or C is bikini top or C is bra:
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:215]: A bikini top is a kind of bra so the middle part of this line is redundant.

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 2]
```diff
@@ -207,6 +207,54 @@ To compute stat healing:
...
+				If C is chestless or fully exposing, decide on 0;
+				If the largeness of breasts <= 2,decide on 0;
+				if C is overdress and support of (C - a bra) is >= 2, decide on 0;
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:218]: Is this `decide on 0;` something you've learned to do elsewhere to break from a block early or something? I can imagine it might work, I've just never tried it before.

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 3]
```diff
@@ -207,6 +207,54 @@ To compute stat healing:
...
 		say "[bold type]You feel as though your mind is slightly clearer![roman type]";
 
+A time based rule (this is the audible jiggling rule):
+	if a random number between 1 and 100 < 50 + unlucky:
+		if debugmode > 0, say "AUDIBLE CHECK. ";
+		let audibly jiggling clothing be nothing;
+		repeat with C running through worn cursed blandness clothing:
+			if C is overdress or C is bikini top or C is bra:
+				If C is chestless or fully exposing, decide on 0;
+				If the largeness of breasts <= 2,decide on 0;
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:217]: missing whitespace between `,` and `decide`

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x (hunk 2 above)
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:218]: This is just a mess!

You can't use `(c - a bra)` in a command like this. The only place it's appropriate to use this type of notation is when declaring a polymorphic function, for example `To decide which number is the initial outrage of (C - a bra):`

Also you can't write `is >=`, you just write `>=`.

Furthermore if `C` is overdress, as the first part of the condition requires, then trying to check the `support` of `C` will trigger errors because `support` is an attribute of bras.

You are probably trying to make it so that bras with a lot of support aren't chosen. For that, you would do

```if C is bra and the support of C >= 2, decide on 0;```

And that's assuming that your `decide on 0;` actually works as intended.

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 4]
```diff
@@ -207,6 +207,54 @@ To compute stat healing:
...
+				if C is plentiful or rare and a random number between 1 and 3 is 2, now audible jiggling clothing is C;
+		repeat with C running through worn audible jiggling clothing:
+			now audibly jiggling clothing is C;
+		
+		To decide which number is the stealth influence of (C - a clothing)
+			if C is bsounding, decide on -6;
+			decide on 0;
+		
+		To compute bsound:
+			repeat with C running through audibly jiggling clothing:
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:228]: You will want this to only be occurring for items the player is actually wearing, so it needs to be 

```repeat with C running through worn audibly jiggling clothing:```

(or `audible jiggling` if we change the name)

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 5]
```diff
@@ -207,6 +207,54 @@ To compute stat healing:
...
+				say "Your [ShortDesc of C] accompanies the bounces of your [ShortDescBreasts] with soft boings and wobbles.[or] Your [C] is ensuring that every [BreastBounceDesc] your [RealBreastDesc]. [or]You are unable to prevent your [C] from letting everyone know each bounce of your [ShortDescBreasts]![or][bold type]The jiggling of your [ShortDescBreasts] is likely to attract others to your location![roman type][line break][or] The noises your [ShortDesc of C] is making signals exactly where you are .[or]You wish you could stop your [C] from detailing your breasts as they [BreastBounceDesc] .[as decreasingly likely outcomes]";
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:229]: Again, the correct terms are either `[ShortDesc of breasts]` or `[BreastDesc]`.

Furthermore there's a couple of times here where you have put whitespace before the `.` instead of after, at the end of the last two segments of your `[one of]` construction.

There's also an unnecessary whitespace before "The noises".

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 6]
```diff
@@ -207,6 +207,54 @@ To compute stat healing:
...
+			now C is bsounding.
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:230]: This should be at the same indentation level as the `say` command above it.

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 7]
```diff
@@ -207,6 +207,54 @@ To compute stat healing:
...
+			
+		bsounding is a humilating situation.
+		Definition: bsounding (Called A) is applicable:
+			if the player is not disgraced and there is a bsound, decide yes;
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:234]: `there is a bsound` is meaningless. I assume what you mean is `there is a worn bsounding clothing`.

We also have a slight clash here, because depending on code compilation order, `the reflection rule` which checks for humiliating situations might occur after the `clothes stops bsounding rule` you have declared below. Which would then mean there was no bsounding clothing again. So what would be better is if instead of having the ``clothes stops bsounding rule` declared as you have below, you instead add the content of that block to the `To Reset Flags` block in `Compute Turn.i7x`.

**Review by aika092 — CHANGES_REQUESTED** [review]
Okay fine you asked for it, here is a full review 😛 

Perhaps now that I've done this, you could try making the required changes yourself?

File: Trap Quest.inform/Inform/Extensions/Clothing/Clothing Adjectives.i7x [hunk 8]
```diff
@@ -253,7 +253,7 @@ Clothing can be ass plugging. Clothing is usually not ass plugging.
 Clothing can be vagina plugging. Clothing is usually not vagina plugging.
 Clothing has a number called plug size. The plug size of clothing is usually 0.
 Clothing can be purity. Clothing is usually not purity. [Means they care about your virginity.]
-A Magic-type is a kind of value. The magic-types are blandness, dressup, milk production, absorption, temptation, suppression, bed wetting, confidence, endurance, dominance, constriction, speed, kicking, protection, posture training, expansion, refreshment, rejuvenation, possession, maturity, respiration, durability, stumbling, provocation, exposure, and hostility. Clothing has a magic-type. The magic-type of clothing is usually blandness.
+A Magic-type is a kind of value. The magic-types are blandness, dressup, milk production, absorption, temptation, suppression, bed wetting, confidence, endurance, dominance, constriction, speed, kicking, protection, posture training, expansion, refreshment, rejuvenation, possession, maturity, respiration, durability, stumbling, provocation, exposure, audibly jiggling, and hostility. Clothing has a magic-type. The magic-type of clothing is usually blandness.
```
[the remark this answers is not in the document]
Comment (InraInaro) [reply Trap Quest.inform/Inform/Extensions/Clothing/Clothing Adjectives.i7x]: I changed it from audible jiggling to audible jiggles. I think audible jiggling is fine, however personally "+1 halloween schoolgirl outfit of audible jiggles" fits better. My grammar knowledge is rather weak though.

File: Trap Quest.inform/Inform/Extensions/Dresses/Cheerleader Outfit.i7x [hunk 9]
```diff
@@ -1,6 +1,6 @@
 Cheerleader Outfit by Dresses begins here.
 
-A cheerleader outfit is a kind of overdress. A cheerleader outfit is unique. A cheerleader outfit is usually short. Understand "cheer", "leader" as cheerleader outfit.
+A cheerleader outfit is a kind of overdress. A cheerleader outfit is unique. A cheerleader outfit is usually short. A cheerleader outfit is usually audibly jiggling. Understand "cheer", "leader" as cheerleader outfit.
```
[the remark this answers is not in the document]
Comment (InraInaro) [reply Trap Quest.inform/Inform/Extensions/Dresses/Cheerleader Outfit.i7x:3]: I had two ideas for a benefit: Enemies would have a higher chance of going for breasts and/or friendly followers would stick around longer. Though I have no idea on how either might be implemented. I don't think that charisma makes much sense with it aimed to be embarrassing, though I guess you could argue against my second benefit similarly. If you implement the first idea, then another smaller benefit might balance out the negatives, though I can't think of one right now. 

A condition that checks the breasts of the player would make it equal to the way it normally spawns. Perhaps adding in a size check during each time it transforms after the fact, it would prevent it from never getting attached. For the schoolgirl outfit I think the unlucky modifier might be a good balance since it's something easily attained. However the cheerleader class is both a rarer class and a derivative from the schoolgirl when you get pigtails, so I don't think it's necessary there.

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x (hunk 2 above)
Comment (InraInaro) [reply Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x]: I saw it used in some of the other files here. I thought it acted similarly to a type of break statement that I know from my rudimentary knowledge of Java.

Comment (InraInaro) [reply Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:218]: I was trying to kill two birds with one stone. I didn't think it would be immersive to have it trigger when there was a bra with a lot of support underneath or for it to be the bra. It's supposed to check for the bra that the player is currently wearing and make sure it doesn't have support more than or equal to 2. I'm going to change the line to what you suggested, though do you have an idea to how that can work?

**Comment by InraInaro** [comment]
Still need to do something with a block in periodic effects and move a code block to the compute turn file.

File: Trap Quest.inform/Inform/Extensions/Dresses/Cheerleader Outfit.i7x (hunk 9 above)
[the remark this answers is not in the document]
Comment (aika092) [reply Trap Quest.inform/Inform/Extensions/Dresses/Cheerleader Outfit.i7x]: Friendly followers actually *will* stay around a bit longer on average with this effect, because they will be able to stalk the player almost flawlessly, and so they won't run the risk of going "hmm I've actually ended up too far away from the player and can't be bothered to keep seeking them out". But yes we could also alter how quickly their "friendly boredom" increases, or what level it needs to reach for them to lose interest... however I would argue that in most cases, friendly NPCs following you around is a bad thing. Since it means that if they see you doing something lewd or private - anything from undressing to using the toilet to masturbating - they will lose favour with you.

The only real time it's good to have someone follow you around is if they love you enough to fight with you.

Increased chance of NPCs going for breasts is easy to implement but arguably only a slight positive, since if too many NPCs titfuck you it's still pretty intense, what with the increased sensation and addiction and eventually orgasms. Also it raises the question of whether perhaps the clothing should become `bsounding` every turn an NPC fucks your breasts! 😄 

If what you're suggesting is that the chance of audible jiggling appearing increases with breast size, I love that and think it's a great idea.

It's not impossible to check after each transformation, too. A little fiddly but can be done. The best way to do it would be to use the `To compute post transformation effect of (C - a clothing):` function which fires after a clothing has been transformed.

So the full block would look something like:

```
To compute post transformation effect of (C - a schoolgirl outfit):
	if a random number between 1 and 20 < the largeness of breasts, now C is audible jiggles.
```

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x (hunk 2 above)
Comment (aika092) [reply Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x]: I use it when the block is a function that returns a number, i.e. starts with `To decide which number is`.

I don't think it's particularly necessary to do this here especially because it might have weird unexpected functionality. There are lots of other ways to use if statements and logic and indentation to make sure that only the desired items reach the next bit of code.

Comment (aika092) [reply Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x]: We'd need to first check for a bra. Since we know that the player can only wear one bra at a time, the shortest way to do it would be like this:

`if there is a worn bra and the support of a random worn bra >= 2`

But I *mostly* tend to avoid doing this because every now and then I have found that the interpreter appears to mess up and not successfully obtain the correct object and return the correct attribute (in this case, `support`). So I prefer, just to be safe, to do it like this:

```
let B be a random worn bra;
if B is bra and the support of B >= 2
```

Having said that I just checked my code and you can see in lines 1258 and 1259 of Crafting\Alchemy Products.i7x there is a (hopefully lol) working example of doing it the first way.

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 10]
```diff
@@ -207,53 +207,49 @@ To compute stat healing:
...
-		To decide which number is the stealth influence of (C - a clothing)
-			if C is bsounding, decide on -6;
-			decide on 0;
-		
-		To compute bsound:
-			repeat with C running through audibly jiggling clothing:
-				say "Your [ShortDesc of C] accompanies the bounces of your [ShortDescBreasts] with soft boings and wobbles.[or] Your [C] is ensuring that every [BreastBounceDesc] your [RealBreastDesc]. [or]You are unable to prevent your [C] from letting everyone know each bounce of your [ShortDescBreasts]![or][bold type]The jiggling of your [ShortDescBreasts] is likely to attract others to your location![roman type][line break][or] The noises your [ShortDesc of C] is making signals exactly where you are .[or]You wish you could stop your [C] from detailing your breasts as they [BreastBounceDesc] .[as decreasingly likely outcomes]";
-			now C is bsounding.
+	To compute bsound:
+		repeat with C running through worn AJC:
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:224]: You've put `AJC` here (and several times below) but this was just a local variable from the `audible jiggling rule`. Here (and every time below) you need to spell out `audible jiggles clothing` like normal.

Also unless this is a glitch in the web interface for GitHub, this still is indented incorrectly, since there is intendation before `To compute bsound:` even though this is the start of a new block.

Same for several blocks below this one.

File: Trap Quest.inform/Inform/Extensions/Dresses/Cheerleader Outfit.i7x (hunk 9 above)
[the remark this answers is not in the document]
Comment (InraInaro) [reply Trap Quest.inform/Inform/Extensions/Dresses/Cheerleader Outfit.i7x]: Having the clothes become bsounding when a titfuck happens would be great. 

Charisma and seductive bonuses are better than my two suggestions with that explanation. I was also thinking a magic bonus but I forget if that would make it too easy to swap over into the magical girl class. 

With this check, should there also be "if largeness of breast < 2, {break statement}" to prevent those with flat chests from acquiring it?

[the remark this answers is not in the document]
Comment (aika092) [reply Trap Quest.inform/Inform/Extensions/Dresses/Cheerleader Outfit.i7x]: If we are making audible jiggles only appear on schoolgirls upon transformation, and with RNG, I think it's fine to leave it without a positive effect. However I also would be fine with it giving a +1 to charisma.

> With this check, should there also be "if largeness of breast < 2, {break statement}" to prevent those with flat chests from acquiring it?

I'm not sure what check you mean? If you mean in the post transformation effect thing, that's already taken care of since a random number between 1 and 20 cannot be lower than 1. If we want to avoid it be lower than 2, since AA cup boobies are probably too small to make any noise, we could just change it to a random number between 2 and 20.

Since we're resetting all `bsounding` flags at the end of the turn, the best time to add a `bsounding` flag during titfucks is probably very early on in the turn, i.e. before monsters have their turn. So I would do this:

```
An all time based rule (this is the audible jiggling from titfucks rule):
	if there is a male monster penetrating breasts:
		repeat with C running through worn audible jiggles clothing:
			now C is bsounding.
```

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 11]
```diff
@@ -221,36 +221,36 @@ A time based rule (this is the audible jiggles rule):
...
-		compute bsound.
+Report jumping when there is a worn audible jiggles clothing:
+	compute bsound.
 
-	An all later time based rule (this is the clothes stops bsounding rule):
-		repeat with A running through bsounding clothing:
-			now A is not bsounding;
+An all later time based rule (this is the clothes stops bsounding rule):
+	repeat with A running through bsounding clothing:
+		now A is not bsounding;
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x:253]: looking great now!

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 12]
```diff
@@ -221,36 +221,36 @@ A time based rule (this is the audible jiggles rule):
...
 			
-	bsoundingHS is a humilating situation.
-	Definition: bsoundingHS (Called A) is applicable:
-		if the player is not disgraced and there is a worn bsounding clothing, decide yes;
+bsoundingHS is a humilating situation.
+Definition: bsoundingHS (Called A) is applicable:
+	if the player is not disgraced and there is a worn bsounding clothing, decide yes;
 		decide no.
 	To reflect on (A - bsoundingHS):
 		say "[first custom style][one of] Who had the bright idea to make clothing that announced every wobble of these damned orbs?! [or]I've got to get dump this [C]. It's impossible to get any long lasting privacy with these stupid boings and twangs![or]Do these [ShortDesc of C] run out of energy? How humiliating![stopping][roman type][line break]".
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x]: From where I'm looking these three lines are still indented one too many times

File: Trap Quest.inform/Inform/Extensions/Every Turn/Compute Turn.i7x [hunk 13]
```diff
@@ -938,7 +938,8 @@ To Reset Flags:
 		if being-fucked is 0, now the buildup of F is 0;
 	if the rawness of penis > 0 and wanking is 0 and the number of things penetrating penis is 0 and the number of things penetrating vagina is 0:
 		RawDown penis.
-
+	repeat with A running through bsounding clothing:
+		now A is not bsounding;
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Compute Turn.i7x:942]: This needs to go before line 939, since the block currently ends at line 939 & 940. So these two lines are just floating in the air at the moment.

File: Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x [hunk 14]
```diff
@@ -211,15 +211,12 @@ A time based rule (this is the compute jiggles rule):
...
-				if B is bra and the support of B >= 2, decide on 0;
-				if (C is plentiful or C is rare) and a random number between 1 and 3 is 2, now AJC is C;
-		repeat with C running through worn AJC:
-			now AJC is C;
+		unless there is worn audible jiggles clothing or the largeness of breasts <= 2 or (B is bra and the support of B >= 2):
+			let AJC be nothing;
+			repeat with C running through worn cursed blandness ridiculously low cut or higher clothing: [ridiculously low cut or higher clothing is essentially any clothing that's not chestless or fully exposing]
+			if AJC is clothing:
+				now AJC is audible jiggles;
+					say "Your [ShortDesc of C] boinks and pings as your [BreastDesc] move! Must be some sort of magic."
```
Comment (aika092) [inline Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x]: Something has gone a bit wrong here. A number of mistakes.

It looks like there is a line missing after line 216. I think it was supposed to say 
```
if (C is plentiful or C is rare) and a random number between 1 and 3 is 2, now AJC is C;
```
obviously with one intendation level more than line 216.

Line 219 (the `say` command) is indented one too many times. It should have the same indentation level as the line above it (`now AJC is audible jiggles;`).


There's also one more thing I've just noticed. You've set up the RNG to trigger every time a number between 1 and 100 is less than 50. That's a huge amount, that means every 2 turns on average, the game will be looking to try and turn a piece of cursed clothing into an audible jiggles item. That's way too frequent. What would make more sense is something like `if a random number between 1 and 100 < 2 + unlucky:` because then it will be 1 in every 100 turns normally, and 1 in every 50 turns for people with the "Fashion Victim" disadvantage. This is more than enough for a random spontaneous event - in fact we may end up needing to reduce it further, or restrict what items it can happen to further. The way you have it set up now, every single cursed breast covering item without a magic-type will be becoming audible jiggles almost instantly, for the whole game!!!

**Comment by aika092** [comment]
Cool cool cool, almost there now, I tried to compile and there were a couple of other small things and one big thing that you couldn't be expected to anticipate (the humiliating situation stuff needed to be moved to a different file so that it was parsed *after* the humiliating situation object was defined and not before), but all sorted now and it looks like it might complete compilation. I'm super tired so going to bed now, but will check in the morning and then commit my changes if it's working.

**Comment by InraInaro** [comment]
> Cool cool cool, almost there now, I tried to compile and there were a couple of other small things and one big thing that you couldn't be expected to anticipate (the humiliating situation stuff needed to be moved to a different file so that it was parsed _after_ the humiliating situation object was defined and not before), but all sorted now and it looks like it might complete compilation. I'm super tired so going to bed now, but will check in the morning and then commit my changes if it's working.

Sorry for the late response. I'm glad to hear that it's close to post. I underestimated how ignorant I was, so thanks for taking the time to explain the errors I made.

**Comment by aika092** [comment]
Copying and pasting my notes for the changes I made to your code here:

* Moved the `compute bsound` calling code for Carry Out Going to the `monsters-go-next rule` to guarantee it is always called before monsters move.
* Established a minimum breast size for audible jiggles stuff to occur throughout the code - largeness 4.
* Cheerleader outfits and schoolgirl outfits operate the same: They have an RNG chance to become audible jiggles, depending on the largeness of breasts.
* The tartan tube top which spawns when the first blue scrunchie is worn also has an RNG-based chance of becoming audible jiggles, scaling with breast size. It's actually more likely than the others, because it caps at max breast size rather than 20.
* The chance of a cursed breast covering item spontaneously becoming audible jiggles now scales logarithmically with breast size.
* Added a function to ensure that boobs always jiggle loudly during every turn of a titfuck.
* Added extra flavour to the `compute bsound` output since this will often occur every single turn, so needs to be nice and varied.
* Ensured that the flavour from `compute bsound` only happens once per turn even if the function is called more than once in a turn.
